From 88c7c1390d4df415fcd8d2bf27c96062e5399830 Mon Sep 17 00:00:00 2001
From: Andrei Stefanescu <andrei.stefanescu@nxp.com>
Date: Wed, 3 Aug 2022 15:02:30 +0300
Subject: [PATCH 049/102] pinctrl: s32: add support for SRE configuration

Add support to configure the SRE property of a MSCR register.

Issue: ALB-8868

Upstream-Status: Pending 

Signed-off-by: Andrei Stefanescu <andrei.stefanescu@nxp.com>
Signed-off-by: Zhantao Tang <zhantao.tang@windriver.com>
---
 drivers/pinctrl/nxp/pinctrl-s32cc.c | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/drivers/pinctrl/nxp/pinctrl-s32cc.c b/drivers/pinctrl/nxp/pinctrl-s32cc.c
index 7ea38f0e71..8478d8ab39 100644
--- a/drivers/pinctrl/nxp/pinctrl-s32cc.c
+++ b/drivers/pinctrl/nxp/pinctrl-s32cc.c
@@ -21,6 +21,8 @@
 #define SIUL2_MSCR_ODE	BIT(20)
 
 #define SIUL2_MSCR_SSS_MASK	0x7
+#define SIUL2_MSCR_SRE_SHIFT	14
+#define SIUL2_MSCR_SRE_MASK	GENMASK(16, 14)
 
 #define UPTR(a) ((uintptr_t)(a))
 
@@ -160,6 +162,12 @@ static int s32_pinconf_set(struct udevice *dev, unsigned int pin_selector,
 	case PIN_CONFIG_DRIVE_PUSH_PULL:
 		mscr_value &= ~SIUL2_MSCR_ODE;
 		break;
+	case PIN_CONFIG_SLEW_RATE:
+		argument = (argument << SIUL2_MSCR_SRE_SHIFT) &
+			   SIUL2_MSCR_SRE_MASK;
+		mscr_value &= ~SIUL2_MSCR_SRE_MASK;
+		mscr_value |= argument;
+		break;
 	default:
 		return -ENOTSUPP;
 	}
-- 
2.17.1

